js/core/utils/array.js

```diff
@@ -3,10 +3,16 @@ import { orderEach } from './object';
 import config from '../config';
 
 function createOccurrenceMap(array) {
-    return array.reduce((map, value) => {
-        map[value] = (map[value] ?? 0) + 1;
-        return map;
-    }, {});
+    return array.reduce((map, value) => map.set(value, (map.get(value) ?? 0) + 1), new Map());
+}
+
+function getFilteredArray(from, toRemove, getUniqueItems) {
+    const toRemoveMap = createOccurrenceMap(toRemove);
+    return from.filter(value => {
+        const occurrencesCount = toRemoveMap.get(value);
+        occurrencesCount && toRemoveMap.set(value, occurrencesCount - 1);
+        return getUniqueItems ^ occurrencesCount;
```

Comment:
To me this function only worsens a readability. Why did u decide to separate it?
- it increases count of code rows
- it needs to pass a non-intuitive true/false parameter
- it uses binary operator
- we don't see simple filtering in parent functions

Reply:
i extracted a function to reuse logic, not sure i follow your first argument - "it increases count of code rows" - actually it does ) i changed it, please take a look

Reply:
oh, Alex, i didn't note that we use Map instead of object => code copypaste is much wide that was before, sorry

Reply:
But i think actual variant is ok anyway